### PR TITLE
docs: update the setup guide

### PR DESCRIPTION
**Pull request by raisedadead**
This PR re-arranges the order for the repository forking and preping the machine. Makes GitPod more prominent and adds warnings about Windows-based setups

### CODE REVIEW

File: docs/how-to-setup-freecodecamp-locally.md [hunk 1]
```diff
@@ -1,44 +1,22 @@
...
-
-This is essential, as it allows you to work on your own copy of freeCodeCamp on GitHub, or to download (clone) your repository to work on locally. Later, you will be able to request changes to be pulled into the main repository from your fork via a pull request (PR).
-
-> [!TIP]
-> The main repository at `https://github.com/freeCodeCamp/freeCodeCamp` is often referred to as the `upstream` repository.
->
-> Your fork at `https://github.com/YOUR_USER_NAME/freeCodeCamp` is often referred to as the `origin` repository. `YOUR_USER_NAME` would be replaced with your GitHub username.
-
-**Follow these steps to fork the `https://github.com/freeCodeCamp/freeCodeCamp` repository:**
+If you are looking to make a one-off contribution, you should use GitPod to make changes. The Gitpod setup launches a ready-to-code environment in a few minutes in your web browser. For contributing long-term, we recommend you setup freeCodeCamp on your local machine.
```
Comment (ShaunSHamilton) [inline docs/how-to-setup-freecodecamp-locally.md:5]:
```suggestion
If you are looking to make a one-off contribution, you should use Gitpod to make changes. The Gitpod setup launches a ready-to-code environment in a few minutes in your web browser. For contributing long-term, we recommend you setup freeCodeCamp on your local machine.
```

File: docs/how-to-setup-freecodecamp-locally.md [hunk 2]
```diff
@@ -1,44 +1,22 @@
...
 
-1. Go to the freeCodeCamp repository on GitHub: <https://github.com/freeCodeCamp/freeCodeCamp>
-
-2. Click the "Fork" Button in the upper right-hand corner of the interface ([More Details Here](https://help.github.com/articles/fork-a-repo/))
+Here are some pros and cons which should help you decide which option is best for you:
 
-3. After the repository has been forked, you will be taken to your copy of the freeCodeCamp repository at `https://github.com/YOUR_USER_NAME/freeCodeCamp` (`YOUR_USER_NAME` would be replaced with your GitHub user name.)
+| GitPod                               | Your own machine (local setup) |
```
Comment (ShaunSHamilton) [inline docs/how-to-setup-freecodecamp-locally.md:9]:
```suggestion
| Gitpod                               | Your own machine (local setup) |
```

File: docs/how-to-setup-freecodecamp-locally.md [hunk 3]
```diff
@@ -1,44 +1,22 @@
...
-
-## Use a Local Machine or GitPod
-
-Once you have forked the repository, you can either use your own local machine or a GitPod-based workspace to work on the codebase.
-
-For contributing long-term, we recommend you setup freeCodeCamp on your local machine.
+> [!ATTENTION]
+> **Note:** If you are using Windows 10 or 11, you will need to use WSL2. You can follow [this guide](how-to-setup-wsl.md) to set up WSL2. You can't use Command Prompt, Git Bash or PowerShell to run freeCodeCamp natively within windows.
 
 ### How to prepare a GitPod workspace
```
Comment (ShaunSHamilton) [inline docs/how-to-setup-freecodecamp-locally.md:21]:
```suggestion
### How to prepare a Gitpod workspace
```

File: docs/how-to-setup-freecodecamp-locally.md [hunk 4]
```diff
@@ -63,13 +41,16 @@ That's it, you can now skip to the 'syncing up from parent' section after you ha
...
+
+- 8 GB RAM
+- Relatively fast CPU (4+ cores)
+- Windows 10 or 11 (with WSL), macOS, or Linux
 
-We primarily support development on Linux and Unix-based systems. Our staff and community contributors regularly work with the codebase using tools installed on Ubuntu and macOS.
+Start by installing the prerequisite software for your operating system.
 
-We also support Windows 10 and 11 via WSL2, which you can prepare by [reading this guide](how-to-setup-wsl.md).
+We primarily support development on Linux and Unix-based systems like Ubuntu and macOS. You can develop on Windows 10 or 11 with WSL2 only. You can follow [this guide](how-to-setup-wsl.md) to set up WSL2. You can't use Command Prompt, Git Bash or PowerShell to run freeCodeCamp natively within windows.
```
Comment (ShaunSHamilton) [inline docs/how-to-setup-freecodecamp-locally.md:52]: I am happy with keeping this (these), but there is repetition between this and the `!ATTENTION` above.

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Last year, I just learnt Gitpod's brand is not _"GitPod"_.

File: docs/how-to-setup-freecodecamp-locally.md (hunk 4 above)
Comment (raisedadead) [reply docs/how-to-setup-freecodecamp-locally.md:52]: Yes - it was intentional for people not reading. In the longer term the UX around the guide can be made nicer. I am working on something on the side.

**Comment by raisedadead** [comment]
Thanks for the sanity checks @ShaunSHamilton

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍 

PS: _there is still at least 1 instance of "GitPod", but it is outside of this PRs changes_
PPS: _i am sad the "how to fork" gif is in light mode_ 😞

**Comment by Sembauke** [comment]
> i am sad the "how to fork" gif is in light mode 

This should definitely be changed 😜

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Can I nitpick that the table be formatted?

**Comment by raisedadead** [comment]
Sure - do you mind pushing a commit? My VS Code settings do not seem to auto-format it, and I will not add spaces manually - lol.

File: docs/how-to-setup-freecodecamp-locally.md [hunk 5]
```diff
@@ -1,46 +1,24 @@
...
 
-3. After the repository has been forked, you will be taken to your copy of the freeCodeCamp repository at `https://github.com/YOUR_USER_NAME/freeCodeCamp` (`YOUR_USER_NAME` would be replaced with your GitHub user name.)
+| Gitpod                               | Your own machine (local setup) |
+| ------------------------------------ | ---------------- |
+| No minimum hardware requirements     | Specific and minimum requirements |
+| No need to install any software      | Additional software required      |
+| Always up to date copy of repository | Need to maintain a local copy of the repository |
+| Slower and can take a few minutes to launch | Faster and can be launched in seconds |
+| Need an internet connection to work  | Minimal internet connection required (once setup) |
+| Some tasks like compilation and tests can take longer to complete | Faster completion of tasks (depending on your machine's capabilities) |
```
Comment (naomi-lgbt) [inline docs/how-to-setup-freecodecamp-locally.md:16]:
```suggestion
| Gitpod                                                            | Your own machine (local setup)                                        |
| ----------------------------------------------------------------- | --------------------------------------------------------------------- |
| No minimum hardware requirements                                  | Specific and minimum requirements                                     |
| No need to install any software                                   | Additional software required                                          |
| Always up to date copy of repository                              | Need to maintain a local copy of the repository                       |
| Slower and can take a few minutes to launch                       | Faster and can be launched in seconds                                 |
| Need an internet connection to work                               | Minimal internet connection required (once setup)                     |
| Some tasks like compilation and tests can take longer to complete | Faster completion of tasks (depending on your machine's capabilities) |
```

Comment (naomi-lgbt) [reply docs/how-to-setup-freecodecamp-locally.md]: I had this in my review I promise GitHub just ate it

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**